test/product-import.spec.coffee

```diff
@@ -537,6 +539,84 @@ describe 'ProductImport unit tests', ->
         expect(err).toBe exptectedErrorMessage
         done()
 
+  describe '::_getExistingProductsForSkus', ->
+
+    it 'should split into multiple querys', (done) ->
+      spyOn(@import.client.productProjections, 'fetch').andReturn({
+        then: (fn) -> fn({ body: { results: [] } })
+      })
+      # 3 bytes string
+      sku = 'SKU'
+      skus = []
+      for i in [1..10000]
+        skus.push(sku)
+      chunks = @import._separateSkusChunksIntoSmallerChunks(skus, skus)
+      # the number of requests should be the same as the number of chunks
+      @import._getExistingProductsForSkus(skus)
+      .then (products) =>
+        actual = @import.client.productProjections.fetch.calls.length
+        expected = chunks.length
+
+        expect(actual).toEqual(expected)
+        done()
+      .finally -> done()
+
+    it 'should accumulate the results of split queries', (done) ->
+      spyOn(@import.client.productProjections, 'fetch').andReturn({
+        then: (fn) -> fn({ body: { results: ['result1', 'result2'] } })
+      })
+      # 3 bytes string
+      sku = 'SKU'
+      skus = []
+      for i in [1..10000]
+        skus.push(sku)
+      chunks = @import._separateSkusChunksIntoSmallerChunks(skus, skus)
+      expectedResult = _.flatten(_.map(chunks, () ->
+        return ['result1', 'result2']
+      ))
+      # the number of requests should be the same as the number of chunks
+      @import._getExistingProductsForSkus(skus)
+      .then (products) ->
+
+        actual = products
+        expected = expectedResult
+
+        expect(actual).toEqual(expected)
+        done()
+      .finally -> done()
+
+  describe '::_separateSkusChunksIntoSmallerChunks', ->
+
+    it 'should split a list of skus that is to big for a single query
+    into chunks that are small enough for a query', ->
+      # prepare fixed bytes
+      whereQuery = "
+        masterVariant(sku in ()) or variants(sku in ())
+      "
+      fixBytes = Buffer.byteLength(encodeURIComponent(whereQuery), 'utf-8')
+      # 3 bytes string
+      sku = 'SKU'
+      skus = []
+      for i in [1..10000]
+        skus.push(sku)
+      # skus is now a 30000 bytes
+      chunks = @import._separateSkusChunksIntoSmallerChunks(skus, skus)
+      # the number of chunks should be at least the number of actual bytes
+      # divided by the number of maximum bytes per request
+      expect(chunks.length).toBeGreaterThan(30000 / 8072)
```

Comment:
Would be great if you could validate if chunks really contain all required SKUs. Otherwise we could end up with faulty product create request for products which has not been found due to accidentally skipped SKUs in chunks logic.